### PR TITLE
Auth: Hide SigV4 config UI and disable middleware when its config flag is disabled

### LINKED ISSUE #37120: SigV4 Auth Details section appears even though `GF_AUTH_SIGV4_AUTH_ENABLED=false`
**What happened**:
The SigV4 Auth Details section appears for existing Prometheus data sources even though `GF_AUTH_SIGV4_AUTH_ENABLED=false`. However, the toggle for SigV4 auth does not appear, as shown in the image below, which is confusing for users.

image

**What you expected to happen**:
When `GF_AUTH_SIGV4_AUTH_ENABLED=false` is set, the toggle for SigV4 auth should not appear **and** the SigV4 Auth Details section should not appear.

**How to reproduce it (as minimally and precisely as possible)**:
1. Install any Grafana version (tested with 7.5.5 and 8.0.6).
2. Set `GF_AUTH_SIGV4_AUTH_ENABLED=true`.
3. Setup a Prometheus data source using SigV4 Auth.
4. Set `GF_AUTH_SIGV4_AUTH_ENABLED=false`.
5. Edit the existing Prometheus data source you created in step 3. The SigV4 toggle should be gone but the SigV4 Auth Details section still exists.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.5.5
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser: MacOS & Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by wbrowne**
**Which issue(s) this PR fixes**:

Fixes #37120

**Special notes for your reviewer**:
This is using the global `setting` struct which is not recommended, so I'm knowingly taking on some debt here. I'm doing this for simplicity now as passing `setting.Cfg` around is a bit messy given how low level this is. I'll resume this once I catch up with @marefr

### CODE REVIEW

File: pkg/models/datasource_cache.go [hunk 1]
```diff
@@ -139,7 +140,7 @@ func (ds *DataSource) HTTPClientOptions() (*sdkhttpclient.Options, error) {
 		}
 	}
 
-	if ds.JsonData != nil && ds.JsonData.Get("sigV4Auth").MustBool(false) {
+	if ds.JsonData != nil && ds.JsonData.Get("sigV4Auth").MustBool(false) && setting.SigV4AuthEnabled {
```
Comment (marefr) [inline pkg/models/datasource_cache.go:143]: Given almost all data sources been converted to use SDK contracts the usage of this should soon be gone.

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/models/datasource_cache.go (hunk 1 above)
Comment (wbrowne) [reply pkg/models/datasource_cache.go:143]: But we will still need to check for whether or not SigV4 config has been enabled, as that value along with the `jsonData` flag needs to be used in combination (due to original issue).

Alternatively, if there's some way of setting the `jsonData. sigV4Auth` to false when the `SigV4AuthEnabled ` config is disabled, or vice-versa

Comment (marefr) [reply pkg/models/datasource_cache.go:143]: Meant that we probably remove all of this code soon hopefully and rely on the HTTP client provider everywhere. 

Yeah probably good to have the check for now. You could do custom things thru provisioning for example.

**Pull request merged**